### PR TITLE
fix `apps.telegram.tasks.send_log_and_actions_message` retrying tasks

### PR DESCRIPTION
**Pull request by joeyorlando**
# What this PR does

It _appears_ like Telegram may have changed one of the error messages they return for `telegram.error.BadRequest`. This _may_ be causing us to infinitely retry some of these tasks.

Previously we were checking for two variants of the same type of error message:
- "Message to reply not found"
- "Replied message not found"

_However_, if I search for the following [in the logs](https://ops.grafana-ops.net/goto/hMgBb8CSR?orgId=1):
```logql
{namespace="amixr-prod"} |~ `(Message to be replied not found|Message to reply not found|Replied message not found)`
````
I _only_ see references to "Message to be replied not found". I have updated references to the former to this new error log message we are seeing.

Also:
- deduplicate some of the words we check for in `telegram.error.BadRequest` and `telegram.error.Unauthorized` into `apps.telegram.client.TelegramClient.BadRequestMessage` and `apps.telegram.client.TelegramClient.UnauthorizedMessage` respectively
- deduplicate some of the wording we use in the `reason` arg passed to `TelegramToUserConnector.create_telegram_notification_error` into `apps.telegram.models.connectors.personal.TelegramToUserConnector.NotificationErrorReason`
- standardize how we check the `message` attribute of `telegram.error.TelegramError`s  into a new `error_message_is` static method on `apps.telegram.client.TelegramClient`
  - previously we would check these error messages in two different ways:
  ```python3
  # style 1
  if "error message to check" in e.message:
    # do something

  # style 2
  if error.message == "error message to check":
    # do something
  ```

## Which issue(s) this PR closes

Closes https://github.com/grafana/oncall-private/issues/2868

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] Added the relevant release notes label (see labels prefixed w/ `release:`). These labels dictate how your PR will
    show up in the autogenerated release notes.

### CODE REVIEW

File: engine/apps/telegram/tasks.py [hunk 1]
```diff
@@ -145,6 +145,8 @@ def send_log_and_actions_message(self, channel_chat_id, group_chat_id, channel_m
             message_type=TelegramMessage.ACTIONS_MESSAGE
         ).exists()
 
+        print(f"YOOOO log_message_sent: {log_message_sent} actions_message_sent: {actions_message_sent} channel_chat_id: {channel_chat_id} group_chat_id: {group_chat_id} alert_group: {alert_group} reply_to_message_id: {reply_to_message_id}")
```
Comment (joeyorlando) [inline engine/apps/telegram/tasks.py:148]: remove when done testing

File: .nvmrc
Comment (joeyorlando) [inline .nvmrc:1]: moved from `grafana-plugin/.nvmrc`

(cc: @grafana/grafana-oncall-frontend)

File: engine/apps/telegram/decorators.py [hunk 2]
```diff
@@ -75,7 +75,7 @@ def decorated(*args, **kwargs):
         try:
             return f(*args, **kwargs)
         except error.BadRequest as e:
-            if "Replied message not found" in e.message:
+            if TelegramClient.error_message_is(e, [TelegramClient.BadRequestMessage.MESSAGE_TO_BE_REPLIED_NOT_FOUND]):
```
Comment (joeyorlando) [inline engine/apps/telegram/decorators.py:78]: one of the main changes

File: engine/apps/telegram/client.py [hunk 3]
```diff
@@ -171,3 +183,7 @@ def _get_message_and_keyboard(
             raise Exception(f"_get_message_and_keyboard with type {message_type} is not implemented")
 
         return text, keyboard
+
+    @staticmethod
+    def error_message_is(error: TelegramError, messages: list[str]) -> bool:
+        return error.message in messages
```
Comment (joeyorlando) [inline engine/apps/telegram/client.py:189]: one of the main changes

File: engine/apps/telegram/tasks.py [hunk 4]
```diff
@@ -165,20 +165,19 @@ def send_log_and_actions_message(self, channel_chat_id, group_chat_id, channel_m
...
+                        TelegramClient.BadRequestMessage.MESSAGE_TO_BE_REPLIED_NOT_FOUND,
+                    ],
+                ):
                     logger.warning(
                         f"Could not send log and actions messages to Telegram group with id {group_chat_id} "
-                        f"due to 'Chat not found'. alert_group {alert_group.pk}"
+                        f"due to '{e.message}'. alert_group {alert_group.pk}"
                     )
                     return
-                elif e.message == "Message to reply not found":
```
Comment (joeyorlando) [inline engine/apps/telegram/tasks.py:174]: another one of the main changes

**Comment by joeyorlando** [comment]
Unfortunately I can't find anything in Telegram's [API docs](https://core.telegram.org/bots/api) regarding the exact wording of these error messages, so sorta shooting from the hip here and hoping this works 🤞

File: engine/apps/telegram/models/connectors/channel.py [hunk 5]
```diff
@@ -123,15 +123,23 @@ def send_alert_group_message(self, alert_group: AlertGroup) -> None:
...
+            if TelegramClient.error_message_is(
+                e,
+                [
+                    TelegramClient.BadRequestMessage.NEED_ADMIN_RIGHTS_IN_THE_CHANNEL,
+                    TelegramClient.BadRequestMessage.CHAT_NOT_FOUND,
+                ],
+            ):
+                if TelegramClient.error_message_is(
+                    e, [TelegramClient.BadRequestMessage.NEED_ADMIN_RIGHTS_IN_THE_CHANNEL]
+                ):
```
Comment (matiasb) [inline engine/apps/telegram/models/connectors/channel.py:135]: Do we need this extra check/if block? Can't we just use the str representation of the error?

**Review by matiasb — APPROVED** [review]

File: engine/apps/telegram/models/connectors/channel.py (hunk 5 above)
Comment (joeyorlando) [reply engine/apps/telegram/models/connectors/channel.py]: good point 👍 I tried to keep the original wording of the error messaging, but probably not a big deal to update this (ie. `"lack of admin rights"` vs `"Need administrator rights in the channel chat"` (which is what `TelegramClient.BadRequestMessage.NEED_ADMIN_RIGHTS_IN_THE_CHANNEL` is))

updated in https://github.com/grafana/oncall/pull/4851/commits/02543461d889ce35235dd40f8c6dbd3b4f064979

**Pull request merged**